### PR TITLE
fix: scrollable speaker dialog + simplify follow-ups

### PR DESCRIPTION
**Pull request by Pendentive**
## Summary

Fixes a user-reported bug where the speaker-confirmation dialog became unusable with 4+ speakers: the **Confirm / Skip / Deep Identify** buttons scrolled off the bottom and could not be pressed. Also bundles the five simplify follow-ups from PR #123's code review.

## Dialog fix — `_ask_speaker_confirmation`

Before: rows packed top-to-bottom into `root` with a fixed height computed from `num_speakers * 70`. When reasoning lines wrapped or height was capped, the button frame (packed last) slid off-screen.

After:
1. A `bottom_panel` is packed with `side=BOTTOM` **first** so Tk reserves space for the progress bar + boundary notice + buttons regardless of row content.
2. The speaker rows now live in a `Canvas` + `Scrollbar` + inner `Frame` scrollable container with `fill="both", expand=True`.
3. Mouse-wheel scrolling is bound only while the pointer is over the canvas so child dropdown widgets keep their own wheel events.
4. `minsize(500, 260)` guarantees the button bar is always reachable even if the user resizes.
5. The dialog thread body is wrapped in `try/except/finally` so `event.set()` always fires — a Tk init crash no longer hangs the meeting pipeline.
6. Removed the 120s silent-abort timeout on `event.wait()` (same pattern as the meeting-name dialog in PR #123): it was dropping speaker confirmation while the dialog was still on screen.

## Simplify follow-ups (from PR #123 review)

| # | Finding | Fix |
|---|---------|-----|
| 1 | `_quit` never logged the exit banner (only `_restart` did) | Explicit `lifecycle.log_exit_banner(logger)` in `_quit` for symmetry |
| 2 | Exit reasons were stringly-typed across 6+ call sites | New `REASON_*` constants in `lifecycle.py`; call sites import them |
| 3 | `capture.discard_streaming` bypassed the package logger with inline `logging.getLogger(...)` | Uses the existing `from .logger import logger`; drops `exists()+stat()` TOCTOU for a single `try/except OSError` |
| 4 | `step_notify` duplicated `ToastListener`'s config-gate logic | New `notifications.is_toast_enabled(event, cfg)` helper; both call sites share the validator |
| 5 | Faulthandler file-retention fix lived in `__main__.main()` only; `worker.py` still had the historic bug | New `crash_diagnostics.install_faulthandler(path)` helper with retained module-scope file. Used by both the tray and worker processes |

## Tests

- `tests/test_lifecycle_reasons.py` — exported constants + default sentinel match
- `tests/test_notifications_helper.py` — `is_toast_enabled` edge cases (list / absent / garbage / None cfg)
- `tests/test_faulthandler_install.py` — retained file, None fallback on bad path

Total: 30 tests, all green.

## Test plan

- [x] `python -m unittest discover tests` — 30 passing
- [x] Package import smoke-test
- [ ] Manual: run a meeting with 5+ speakers, resize the dialog down to 300px tall, confirm Confirm button is still reachable via scroll
- [ ] Manual: observe one meeting completion, confirm `=== WhisperSync exiting === reason=user_quit` appears on tray Quit

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: whisper_sync/crash_diagnostics.py [hunk 1]
```diff
@@ -4,11 +4,62 @@
...
+        except Exception:
+            pass
+        return None
+    _FAULTHANDLER_FILE = f
+    try:
+        faulthandler.enable(file=f, all_threads=True)
+    except Exception:
+        # Extremely unusual; leave the file retained so any later attempt
+        # to enable can still target it, but don't hide the failure.
+        return None
```
[the remark this answers is not in the document]
Comment (Pendentive) [reply whisper_sync/crash_diagnostics.py]: Fixed. install_faulthandler now closes the retained file and clears _FAULTHANDLER_FILE on enable() failure, then best-effort falls back to faulthandler.enable(all_threads=True) on stderr. The open-failure branch uses the same all_threads=True fallback for parity. Commit ccd21b0.

File: whisper_sync/worker.py [hunk 2]
```diff
@@ -88,8 +88,17 @@ def worker_main(request_queue, response_queue, cfg_snapshot: dict,
...
+    # so the file handle is retained module-scope; otherwise CPython can
+    # GC it and native crash dumps from CTranslate2/CUDA vanish. The log
+    # path mirrors the main process so both streams land in the same
+    # dated file.
+    try:
+        from .crash_diagnostics import install_faulthandler
+        from .logger import get_log_path
+        install_faulthandler(get_log_path())
+    except Exception:
+        faulthandler.enable()  # best-effort fallback to stderr
```
[the remark this answers is not in the document]
Comment (Pendentive) [reply whisper_sync/worker.py]: Fixed: worker.py fallback now calls faulthandler.enable(all_threads=True). Commit ccd21b0.

File: tests/test_faulthandler_install.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    def test_returns_none_on_bad_path_and_falls_back(self):
+        from whisper_sync import crash_diagnostics
+        # A path inside a nonexistent parent directory cannot be opened
+        # "a" mode, which forces the fallback branch.
+        bad = Path("/this/path/should/not/exist/fh.log")
+        result = crash_diagnostics.install_faulthandler(bad)
+        try:
+            self.assertIsNone(result)
+        finally:
+            crash_diagnostics._reset_faulthandler_for_tests()
```
[the remark this answers is not in the document]
Comment (Pendentive) [reply tests/test_faulthandler_install.py]: Fixed with your suggested tempdir-based construction so the bad path is deterministic across platforms. Commit ccd21b0.

File: whisper_sync/__main__.py [hunk 4]
```diff
@@ -1469,23 +1463,64 @@ def _show():
...
             tk.Label(header, text="\U0001f3a4", font=("Segoe UI", 13), bg=bg).pack(side=tk.LEFT)
             tk.Label(header, text="  Identify Speakers", font=("Segoe UI", 11, "bold"),
                      bg=bg, fg=fg).pack(side=tk.LEFT)
 
-            # Speaker rows
+            # Bottom panel: progress + boundary notice + buttons. These are
+            # packed NOW (before the scrollable middle) with side=BOTTOM so
+            # Tk's pack algorithm reserves space for them regardless of how
+            # many speakers are added — the Confirm button never slides off.
+            bottom_panel = tk.Frame(root, bg=bg)
```
[the remark this answers is not in the document]
Comment (Pendentive) [reply whisper_sync/__main__.py]: Removed all em dashes I introduced across this PR (8 sites in __main__.py and crash_diagnostics.py), replaced with colons/commas/rephrasing. Commit ccd21b0.

File: whisper_sync/crash_diagnostics.py [hunk 5]
```diff
@@ -4,11 +4,62 @@
...
 import threading
 import traceback
+from pathlib import Path
+
+# Module-level reference to the faulthandler log file. Without a strong ref,
+# the FileIO can be garbage-collected, its fd closes, and later native crash
+# dumps are silently lost — which is exactly what produced the historic
+# silent-death logs with no forensic trace. Both the main tray process and
+# the worker subprocess install their faulthandler via this module so the
+# fix applies uniformly.
```
[the remark this answers is not in the document]
Comment (Pendentive) [reply whisper_sync/crash_diagnostics.py:20]: Removed the em dash in the module docstring and the retention comment. Commit ccd21b0.

File: whisper_sync/__main__.py [hunk 6]
```diff
@@ -1469,23 +1463,64 @@ def _show():
             conf_colors = {"high": green, "medium": yellow, "low": red}
 
             num_speakers = len(speaker_map)
-            # Each speaker row ~44px + reasoning line ~22px + padding
             screen_h = root.winfo_screenheight()
             max_h = min(700, int(screen_h * 0.8))
+            # Size to content, capped at max_h. When content exceeds the cap,
+            # the middle scrolls (see _make_scrollable below) and the bottom
+            # button bar remains visible because it is packed with side=BOTTOM
+            # BEFORE the scrollable middle.
```
[the remark this answers is not in the document]
Comment (Pendentive) [reply whisper_sync/__main__.py]: Fixed. Comment rewritten to describe the inline Canvas/Scrollbar setup (as you suggested). Commit ccd21b0.
